### PR TITLE
[IND-32] switch order fill handler to onchain translation layer logic

### PR DESCRIPTION
**Pull request by dydxwill**

### CODE REVIEW

File: indexer/services/ender/src/validators/order-fill-validator.ts [hunk 1]
```diff
@@ -83,7 +85,7 @@ export class OrderFillValidator extends Validator<OrderFillEventV1> {
           this.block,
           indexerTendermintEvent,
           txId,
-          orderFillEventWithLiquidity,
+          orderFillEventV1ToOrderFill(orderFillEventWithLiquidity),
```
Comment (vincentwschau) [inline indexer/services/ender/src/validators/order-fill-validator.ts:88]: nit: The return type for `createHandlers` should be `Handler<OrderFillWithLiquidity>` and not `Handler<OrderFillEventV1>`.

File: indexer/services/ender/src/validators/order-fill-validator.ts [hunk 2]
```diff
@@ -10,11 +10,13 @@ import _ from 'lodash';
...
 import { LiquidationHandler } from '../handlers/order-fills/liquidation-handler';
 import { OrderHandler } from '../handlers/order-fills/order-handler';
+import { orderFillEventV1ToOrderFill } from '../helpers/translation-helper';
+import { OrderFillWithLiquidity } from '../lib/translated-types';
 import { OrderFillEventWithLiquidity } from '../lib/types';
 import { validateOrderAndReturnErrorMessage } from './helpers';
 import { Validator } from './validator';
 
-export class OrderFillValidator extends Validator<OrderFillEventV1> {
+export class OrderFillValidator extends Validator<OrderFillWithLiquidity> {
```
Comment (vincentwschau) [inline indexer/services/ender/src/validators/order-fill-validator.ts:19]: nit: Shouldn't this validator still be validating `OrderFillEventV1` objects and not `OrderFillWithLiquidity`?

File: indexer/services/ender/src/lib/translated-types.ts [hunk 3]
```diff
@@ -1,7 +1,34 @@
...
+  /**
+   * Taker fee in USDC quantums. If the taker order is a liquidation, then this
+   * represents the special liquidation fee, not the standard taker fee.
+   */
+  takerFee: Long;
+  /** Total filled of the maker order in base quantums. */
+  totalFilledMaker: Long;
+  /** Total filled of the taker order in base quantums. */
+  totalFilledTaker: Long;
+  liquidity: Liquidity;
```
Comment (vincentwschau) [inline indexer/services/ender/src/lib/translated-types.ts:33]: nit: Add comment stating this is the liquidity of the order in the match to process in the handler.

File: indexer/services/ender/src/lib/translated-types.ts [hunk 4]
```diff
@@ -1,7 +1,34 @@
-import { IndexerAssetPosition, IndexerPerpetualPosition, IndexerSubaccountId } from '@dydxprotocol-indexer/v4-protos';
+import { Liquidity } from '@dydxprotocol-indexer/postgres';
+import {
+  IndexerAssetPosition,
+  IndexerOrder,
+  IndexerPerpetualPosition,
+  IndexerSubaccountId, LiquidationOrderV1,
+} from '@dydxprotocol-indexer/v4-protos';
+import { Long } from '@dydxprotocol-indexer/v4-protos/build/codegen/helpers';
```
Comment (vincentwschau) [inline indexer/services/ender/src/lib/translated-types.ts:9]: nit: Maybe a comment here stating these are the canonical object types for handling messages from the protocol in `ender`.

File: indexer/services/ender/src/validators/order-fill-validator.ts (hunk 1 above)
Comment (dydxwill) [reply indexer/services/ender/src/validators/order-fill-validator.ts:88]: done

File: indexer/services/ender/src/validators/order-fill-validator.ts (hunk 2 above)
Comment (dydxwill) [reply indexer/services/ender/src/validators/order-fill-validator.ts:19]: yes, changed

File: indexer/services/ender/src/lib/translated-types.ts (hunk 3 above)
Comment (dydxwill) [reply indexer/services/ender/src/lib/translated-types.ts:33]: done

File: indexer/services/ender/src/lib/translated-types.ts (hunk 4 above)
Comment (dydxwill) [reply indexer/services/ender/src/lib/translated-types.ts:9]: done

File: indexer/services/ender/src/lib/translated-types.ts (hunk 3 above)
Comment (vincentwschau) [reply indexer/services/ender/src/lib/translated-types.ts:36]: nit: Change the comment structure to use `/* ... */` to match the above comments.

**Review by vincentwschau — APPROVED** [review]

**Pull request merged**